src/classifier/classifier.rs

```diff
@@ -53,6 +53,22 @@ where
     }
 }
 
+fn classify_separator(c: char) -> Option<SeparatorKind> {
+    match deunicode_char(c)?.chars().next()? {
+        // Prevent deunicoding cyrillic chars (e.g. ь -> ' is incorrect)
+        _ if ('\u{0410}'..='\u{044f}').contains(&c) => None, // russian cyrillic letters [а-яА-Я]
+        _ if c == '\u{00a0}' => None,                        // non-breaking space
```

Comment:
I would prefer that we consider this non-breaking space a real space for the engine to match more things.
https://en.wikipedia.org/wiki/Non-breaking_space